monai/networks/layers/simplelayers.py

```diff
@@ -137,6 +138,26 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
         raise NotImplementedError(f"Unsupported mode {self.mode}.")
 
 
+class SkipConnectionWithIdx(SkipConnection):
+    """
+    Combine the forward pass input with the result from the given submodule::
+
+    --+--submodule--o--
+      |_____________|
+
+    The available modes are ``"cat"``, ``"add"``, ``"mul"``.
+    Defaults to "cat" and dimension 1.
+    Inherits from SkipConnection but provides the indizes with each forward pass.
+
+    """
+
+    def __init__(self, submodule, dim: int = 1, mode: Union[str, SkipMode] = "cat") -> None:
+        super().__init__(submodule, dim=dim, mode=mode)
+
+    def forward(self, input, indices):
+        return super().forward(input), indices
```

Comment:
This should go with the Quicknat definition as well, but you can also implement it without the constructor: 

```suggestion
class SkipConnectionWithIdx(SkipConnection):
    """
    Combine the forward pass input with the result from the given submodule::

    --+--submodule--o--
      |_____________|

    The available modes are ``"cat"``, ``"add"``, ``"mul"``.
    Defaults to "cat" and dimension 1.
    Inherits from SkipConnection but provides the indizes with each forward pass.

    """
    def forward(self, input, indices):
        return super().forward(input), indices
```